app/components/SearchableMetricSelector.js

```diff
@@ -1,25 +1,75 @@
 import { useEffect, useState } from 'react'
 
-function Modal(props){
-  const { supportedMetrics, onClose, selectMetric, show } = props;
-  const [searchText, setSearchText] = useState("")
-  if(!show){
-    return null;
+function sortDictionary(dictionary){
+    var sortedTagsArr = Object.entries(dictionary).sort((a,b) => b[1].length - a[1].length)
```

Comment:
Add a comment that Object.entries gives each element as an array of [key, value], so you use b[1] to get the value, which is the array of metric objects, then sort based on number of matching metrics in each group.